packages/hawtio/src/plugins/camel/routes/RouteDiagram.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { CamelContext } from '@hawtiosrc/plugins/camel/context'
+import ReactFlow, {
+  addEdge,
+  ConnectionLineType,
+  Edge,
+  Handle,
+  Node,
+  NodeProps,
+  NodeToolbar,
+  Position,
+  useEdgesState,
+  useNodesState,
+} from 'reactflow'
+import dagre from 'dagre'
+import { CamelNodeData, routeVizualisationService } from '@hawtiosrc/plugins/camel/routes/visualization-service'
+import 'reactflow/dist/style.css'
+import './routeDiagram.css'
+import { Flex, FlexItem, Page } from '@patternfly/react-core'
+
+const dagreGraph = new dagre.graphlib.Graph()
+dagreGraph.setDefaultEdgeLabel(() => ({}))
+
+const nodeWidth = 300
+const nodeHeight = 86
+
+export type ICamelNode = Node & {
+  data: CamelNodeData
+}
+
+const getLayoutedElements = (nodes: ICamelNode[], edges: Edge[], direction = 'TB') => {
+  const isHorizontal = direction === 'LR'
+  dagreGraph.setGraph({ rankdir: direction })
+
+  nodes.forEach(node => {
+    dagreGraph.setNode(node.id, { width: nodeWidth, height: nodeHeight })
+  })
+
+  edges.forEach(edge => {
+    dagreGraph.setEdge(edge.source, edge.target)
+  })
+  dagre.layout(dagreGraph)
+
+  nodes.forEach(node => {
+    const nodeWithPosition = dagreGraph.node(node.id)
+    node.targetPosition = isHorizontal ? Position.Left : Position.Top
+    node.sourcePosition = isHorizontal ? Position.Right : Position.Bottom
+
+    // We are shifting the dagre node position (anchor=center center) to the top left
+    // so it matches the React Flow node anchor point (top left).
+    node.position = {
+      x: nodeWithPosition.x - nodeWidth / 2,
+      y: nodeWithPosition.y - nodeHeight / 2,
+    }
+
+    return node
+  })
+
+  return { nodes, edges }
+}
+const CamelNode = ({ data }: NodeProps<CamelNodeData>) => {
```

Comment:
For the top-level React components, I'd like to declare type `React.FunctionComponent` so that readers can easily find they are React components .